src/user/approval.js

```diff
@@ -141,4 +147,15 @@ module.exports = function (User) {
 		const uids = await User.getUidsFromSet('ip:' + user.ip + ':uid', 0, -1);
 		user.ipMatch = await User.getUsersFields(uids, ['uid', 'username', 'picture']);
 	}
+
+	User.autoApprove = async function () {
+		if (meta.config.autoApproveTime <= 0) {
+			return;
+		}
+		const users = await db.getSortedSetRevRangeWithScores('registration:queue', 0, -1);
+		const now = Date.now();
+		await Promise.all(users
```

Comment:
I think it's better to use a regular for loop here so users are created in the order they are added to the registration queue. Also running many parallel user create functions might cause issues.